### PR TITLE
IA-937 attempt to add automation as submodule

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-937

Note after this PR, you don't need to open another IJ for working on automation tests

Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by Qi77Qi** [comment]
jenkins retest

File: Dockerfile-tests [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM bigtruedata/sbt
```
Comment (rtitle) [inline Dockerfile-tests:1]: Do we need a `Dockerfile-tests` if it's all in the same sbt project? :)

File: automation-tests-docker-entrypoint.sh [hunk 2]
```diff
@@ -5,7 +5,9 @@ echo $SBT_CMD
 
 set -o pipefail
 
-sbt -batch -Djsse.enableSNIExtension=false -Dheadless=true "${SBT_CMD}"
+cp -rf src/test/resources/* automation/src/test/resources
```
Comment (rtitle) [inline automation-tests-docker-entrypoint.sh:8]: I'm curious why this `cp` is needed... though I'm just starting reviewing this PR.

File: Dockerfile-tests (hunk 1 above)
Comment (Qi77Qi) [reply Dockerfile-tests:1]: I think we do unless we revamp how we run automation tests in ci, which I think might not be a bad idea, but I think that's another ticket

File: automation-tests-docker-entrypoint.sh (hunk 2 above)
Comment (Qi77Qi) [reply automation-tests-docker-entrypoint.sh:8]: the render config script in firecloud-automation-tests render them under `src/test/resources/*`, but automation being a sub project, we need these config files under `automation/src/test/resources` now

File: docker/build.sh [hunk 3]
```diff
@@ -224,7 +222,7 @@ function docker_cmd()
 #            # pushes the juptyer notebooks and rstudio docker images that go on dataproc clusters
 #            bash ./docker/build-tool.sh --push -i jupyter -r "${NOTEBOOK_REPO}" -t "${DOCKER_TAG}" -b "${GIT_BRANCH}"
 #            bash ./docker/build-tool.sh --push -i rstudio -r "${NOTEBOOK_REPO}" -t "${DOCKER_TAG}" -b "${GIT_BRANCH}"
-
```
Comment (rtitle) [inline docker/build.sh:227]: Wonder if we should delete the above lines and the `build-tool.sh` script. Maybe in a separate PR.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/ProxyDAO.scala [hunk 4]
```diff
@@ -1,4 +1,5 @@
-package org.broadinstitute.dsde.workbench.leonardo.dao
+package org.broadinstitute.dsde.workbench.leonardo
+package dao
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/ProxyDAO.scala:2]: Is this correct? Several instances below too.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala [hunk 5]
```diff
@@ -378,9 +378,11 @@ class ClusterMonitorActor(val cluster: Cluster,
     val result = cluster.dataprocInfo.stagingBucket match {
       case Some(stagingBucketName) => {
         for {
+          _ <- IO(logger.info("persisting errors")).unsafeToFuture()
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala]: I like to put `${cluster.projectNameString}` in every log message, if I can -- helps with traceability

**Comment by rtitle** [comment]
I think this looks good - I don't really see any functional changes, just refactoring.

I think I like the `http` and `automation` module names.

Sorry I merged a PR and you now have some conflicts. :( Hopefully not to hard to resolve. This PR will be annoying to leave open for an extended time.

I've just scanned GitHub so far but I'll also check out this branch and poke around running unit tests, etc. I'll thumb if all seems good.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala (hunk 5 above)
Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala]: these log lines weren't there before...I'll remove them for now....regarding your comment, I think it's a great point, and I think we should do https://broadworkbench.atlassian.net/browse/IA-906 for adding logging in json format (with clusterName, and googleProject) which can be indexed in kibana

**Review by rtitle — APPROVED** [review]
This seems to work for me. :shipit: once conflicts are resolved, etc.

Might want to give the team a heads up too, may need to re-import in Intellij and change sbt commands :)

**Pull request merged**